Author: Guido Guenther <agx@sigxcpu.org>
Description: No need to check for lib64 (Closes: #565508).

diff -Naurp plymouth.orig/scripts/plymouth-set-default-theme.in plymouth/scripts/plymouth-set-default-theme.in
--- plymouth.orig/scripts/plymouth-set-default-theme.in	2010-01-02 11:22:54.000000000 +0100
+++ plymouth/scripts/plymouth-set-default-theme.in	2010-03-14 17:03:29.000000000 +0100
@@ -5,11 +5,7 @@ set -e
 [ -z "$LIBEXECDIR" ] && LIBEXECDIR="/usr/libexec"
 [ -z "$DATADIR" ] && DATADIR="/usr/share"
 if [ -z "$LIB" ]; then
-  if $(echo nash-showelfinterp /proc/$$/exe | /sbin/nash --forcequiet | grep -q lib64); then
-    LIB="lib64"
-  else
     LIB="lib"
-  fi
 fi
 [ -z "$LIBDIR" ] && LIBDIR="/usr/$LIB"
 [ -z "$BINDIR" ] && BINDIR="/usr/bin"
