Author: Guido Guenther <agx@sigxcpu.org>
Description: Disable noveau.

diff -Naurp plymouth.orig/configure plymouth/configure
--- plymouth.orig/configure	2010-03-27 16:29:57.000000000 +0100
+++ plymouth/configure	2010-03-28 00:27:52.000000000 +0100
@@ -13041,12 +13041,12 @@ if test -n "$DRM_CFLAGS"; then
     pkg_cv_DRM_CFLAGS="$DRM_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libdrm libdrm_intel libdrm_radeon libdrm_nouveau\"") >&5
-  ($PKG_CONFIG --exists --print-errors "libdrm libdrm_intel libdrm_radeon libdrm_nouveau") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libdrm libdrm_intel libdrm_radeon\"") >&5
+  ($PKG_CONFIG --exists --print-errors "libdrm libdrm_intel libdrm_radeon") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_DRM_CFLAGS=`$PKG_CONFIG --cflags "libdrm libdrm_intel libdrm_radeon libdrm_nouveau" 2>/dev/null`
+  pkg_cv_DRM_CFLAGS=`$PKG_CONFIG --cflags "libdrm libdrm_intel libdrm_radeon" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -13057,12 +13057,12 @@ if test -n "$DRM_LIBS"; then
     pkg_cv_DRM_LIBS="$DRM_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libdrm libdrm_intel libdrm_radeon libdrm_nouveau\"") >&5
-  ($PKG_CONFIG --exists --print-errors "libdrm libdrm_intel libdrm_radeon libdrm_nouveau") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libdrm libdrm_intel libdrm_radeon\"") >&5
+  ($PKG_CONFIG --exists --print-errors "libdrm libdrm_intel libdrm_radeon") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_DRM_LIBS=`$PKG_CONFIG --libs "libdrm libdrm_intel libdrm_radeon libdrm_nouveau" 2>/dev/null`
+  pkg_cv_DRM_LIBS=`$PKG_CONFIG --libs "libdrm libdrm_intel libdrm_radeon" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -13080,14 +13080,14 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        DRM_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "libdrm libdrm_intel libdrm_radeon libdrm_nouveau" 2>&1`
+	        DRM_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "libdrm libdrm_intel libdrm_radeon" 2>&1`
         else
-	        DRM_PKG_ERRORS=`$PKG_CONFIG --print-errors "libdrm libdrm_intel libdrm_radeon libdrm_nouveau" 2>&1`
+	        DRM_PKG_ERRORS=`$PKG_CONFIG --print-errors "libdrm libdrm_intel libdrm_radeon" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$DRM_PKG_ERRORS" >&5
 
-	{ { $as_echo "$as_me:$LINENO: error: Package requirements (libdrm libdrm_intel libdrm_radeon libdrm_nouveau) were not met:
+	{ { $as_echo "$as_me:$LINENO: error: Package requirements (libdrm libdrm_intel libdrm_radeon) were not met:
 
 $DRM_PKG_ERRORS
 
@@ -13098,7 +13098,7 @@ Alternatively, you may set the environme
 and DRM_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 " >&5
-$as_echo "$as_me: error: Package requirements (libdrm libdrm_intel libdrm_radeon libdrm_nouveau) were not met:
+$as_echo "$as_me: error: Package requirements (libdrm libdrm_intel libdrm_radeon) were not met:
 
 $DRM_PKG_ERRORS
 
diff -Naurp plymouth.orig/configure.ac plymouth/configure.ac
--- plymouth.orig/configure.ac	2010-03-27 16:29:57.000000000 +0100
+++ plymouth/configure.ac	2010-03-28 00:27:52.000000000 +0100
@@ -56,7 +56,7 @@ PKG_CHECK_MODULES(GTK, [gtk+-2.0 >= 2.12
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-PKG_CHECK_MODULES(DRM, [libdrm libdrm_intel libdrm_radeon libdrm_nouveau])
+PKG_CHECK_MODULES(DRM, [libdrm libdrm_intel libdrm_radeon])
 AC_SUBST(DRM_CFLAGS)
 AC_SUBST(DRM_LIBS)
 
diff -Naurp plymouth.orig/src/plugins/renderers/drm/Makefile.am plymouth/src/plugins/renderers/drm/Makefile.am
--- plymouth.orig/src/plugins/renderers/drm/Makefile.am	2010-01-14 23:05:05.000000000 +0100
+++ plymouth/src/plugins/renderers/drm/Makefile.am	2010-03-28 00:27:52.000000000 +0100
@@ -20,8 +20,6 @@ drm_la_SOURCES = $(srcdir)/plugin.c
 		 $(srcdir)/ply-renderer-i915-driver.h                         \
 		 $(srcdir)/ply-renderer-i915-driver.c                         \
 		 $(srcdir)/ply-renderer-radeon-driver.h                       \
-		 $(srcdir)/ply-renderer-radeon-driver.c                       \
-		 $(srcdir)/ply-renderer-nouveau-driver.h                      \
-		 $(srcdir)/ply-renderer-nouveau-driver.c
+		 $(srcdir)/ply-renderer-radeon-driver.c
 
 MAINTAINERCLEANFILES = Makefile.in
diff -Naurp plymouth.orig/src/plugins/renderers/drm/Makefile.in plymouth/src/plugins/renderers/drm/Makefile.in
--- plymouth.orig/src/plugins/renderers/drm/Makefile.in	2010-03-24 22:06:29.000000000 +0100
+++ plymouth/src/plugins/renderers/drm/Makefile.in	2010-03-28 00:27:52.000000000 +0100
@@ -74,8 +74,7 @@ drm_la_DEPENDENCIES = $(am__DEPENDENCIES
 	../../../libply-splash-core/libply-splash-core.la
 am_drm_la_OBJECTS = drm_la-plugin.lo \
 	drm_la-ply-renderer-i915-driver.lo \
-	drm_la-ply-renderer-radeon-driver.lo \
-	drm_la-ply-renderer-nouveau-driver.lo
+	drm_la-ply-renderer-radeon-driver.lo
 drm_la_OBJECTS = $(am_drm_la_OBJECTS)
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
@@ -275,9 +274,7 @@ drm_la_SOURCES = $(srcdir)/plugin.c
 		 $(srcdir)/ply-renderer-i915-driver.h                         \
 		 $(srcdir)/ply-renderer-i915-driver.c                         \
 		 $(srcdir)/ply-renderer-radeon-driver.h                       \
-		 $(srcdir)/ply-renderer-radeon-driver.c                       \
-		 $(srcdir)/ply-renderer-nouveau-driver.h                      \
-		 $(srcdir)/ply-renderer-nouveau-driver.c
+		 $(srcdir)/ply-renderer-radeon-driver.c
 
 MAINTAINERCLEANFILES = Makefile.in
 all: all-am
@@ -356,7 +353,6 @@ distclean-compile:
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/drm_la-plugin.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/drm_la-ply-renderer-i915-driver.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/drm_la-ply-renderer-nouveau-driver.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/drm_la-ply-renderer-radeon-driver.Plo@am__quote@
 
 .c.o:
diff -Naurp plymouth.orig/src/plugins/renderers/drm/plugin.c plymouth/src/plugins/renderers/drm/plugin.c
--- plymouth.orig/src/plugins/renderers/drm/plugin.c	2010-03-27 16:29:57.000000000 +0100
+++ plymouth/src/plugins/renderers/drm/plugin.c	2010-03-28 00:29:06.000000000 +0100
@@ -454,11 +454,13 @@ load_driver (ply_renderer_backend_t *bac
       backend->driver_interface = ply_renderer_radeon_driver_get_interface ();
       backend->driver_supports_mapping_console = false;
     }
+/*
   else if (strcmp (driver_name, "nouveau") == 0)
     {
       backend->driver_interface = ply_renderer_nouveau_driver_get_interface ();
       backend->driver_supports_mapping_console = false;
     }
+*/
   free (driver_name);
 
   if (backend->driver_interface == NULL)
