--- a/src/libply-splash-core/ply-terminal.c
+++ b/src/libply-splash-core/ply-terminal.c
@@ -136,6 +136,14 @@ ply_terminal_parse_keymap_conf (ply_term
                 keymap = ply_key_file_get_value (vconsole_conf, NULL, "KEYMAP");
         ply_key_file_free (vconsole_conf);
 
+        /* Fallback to the debian specific keyboard configuration file */
+        if (!keymap) {
+                vconsole_conf = ply_key_file_new ("/etc/default/keyboard");
+                if (ply_key_file_load_groupless_file (vconsole_conf))
+                        keymap = ply_key_file_get_value (vconsole_conf, NULL, "XKBLAYOUT");
+                ply_key_file_free (vconsole_conf);
+        }
+
         /* The keymap name in vconsole.conf might be quoted, strip these */
         if (keymap && keymap[0] == '"' && keymap[strlen (keymap) - 1] == '"') {
                 old_keymap = keymap;
